Forte.EpiServer.AzureSearch/Query/Extensions/FilterByAccessExtensions.cs

```diff
@@ -24,5 +31,25 @@ public static AzureSearchQueryBuilder FilterOnRoleAccess(this AzureSearchQueryBu
 
             return queryBuilder.Filter(new FilterComposite(Operator.Or, rolesAccessFilters));
         }
+
+        /// <summary>
+        /// Method filters search items by allowed roles for current user.
+        /// </summary>
+        /// <param name="queryBuilder"></param>
+        /// <returns></returns>
+        public static AzureSearchQueryBuilder FilterOnRoleAccessForCurrentPrincipal(this AzureSearchQueryBuilder queryBuilder)
+        {
+            var currentPrincipal = ServiceLocator.Current.TryGetExistingInstance<IPrincipalAccessor>(out var instance) ? instance.Principal : FallbackPrincipal.Current;
```

Comment:
same here, I'd rather construct anonymous principal manually. Maybe we can extract this (constructing anonymous principal) to some static class and field and use it here and also in XhtmlStringExtractor?